src/backend/distributed/planner/recursive_planning.c

```diff
@@ -359,6 +364,18 @@ RecursivelyPlanSubqueriesAndCTEs(Query *query, RecursivePlanningContext *context
 		RecursivelyPlanLocalTableJoins(query, context);
 	}
 
+	/*
+	 * Similarly, logical planner cannot handle outer joins when the outer rel
+	 * is recurring, such as "<recurring> LEFT JOIN <distributed>". In that case,
+	 * we convert distributed table into a subquery and recursively plan inner
+	 * side of the outer join. That way, inner rel gets converted into an intermediate
+	 * result and logical planner can handle the new query since it's of the from
+	 * "<recurring> LEFT JOIN <recurring>".
+	 *
+	 * See DeferredErrorIfUnsupportedRecurringTuplesJoin for the supported join
+	 * types.
+	 */
+	RecursivelyPlanRecurringTupleOuterJoins((Node *) query->jointree, query, context);
```

Comment:
We should probably change the error detail to mention about correlation:
```SQL
SELECT
	count(*)
FROM
 test_r_1 LEFT JOIN LATERAL
	(SELECT
		*
	FROM
		test_2 WHERE
		test_2.key = test_r_1.key) as foo
ON (true);
ERROR:  cannot pushdown the subquery
DETAIL:  There exist a reference table in the outer part of the outer join

```
such as like:
```C
		preconditionsSatisfied = false;
		errorDetail = psprintf("Offset clause is currently unsupported when a %ssubquery "
							   "references a column from %s", lateralString,
							   referencedThing);
````